### ISSUE TITLE
Missing `NU_LIB_DIRS` on non-interactive use

### ISSUE THREAD

**Issue by tingerrr** [nushell/nushell#9863]
### Describe the bug

The environment variable `$env.NU_LIB_DIRS` does not seem to be populated in non-interactive use (`nu ...scripts`/`nu -c '...'`), the `-I` option is also frequently ignored on both `0.82.0` and `0.83.0` when not passing `-n` (this should if intended at least warn the user).

To me, only the outputs of row `4` and `5` appear sensible in both versions.
It seems the `env.nu` file is not being evaluated anymore.

### How to reproduce

Run the following versions with the given arguments: `cargo run --release -- {args}` and a `test.nu` file containing the line `$env.NU_LIB_DIRS`. Observe the behavior described in the table.

|n|arguments|`0.82.0`|`0.83.0`|
|---|---|---|---|
|0|`-I test -c '$env.NU_LIB_DIRS'`|`[~/.config/nushell/scripts]`|`empty list`|
|1|`        -c '$env.NU_LIB_DIRS'`|`[~/.config/nushell/scripts]`|`empty list`|
|2|`-I test test.nu`|`[~/.config/nushell/scripts]`|`empty list`|
|3|`        test.nu`|`[~/.config/nushell/scripts]`|`empty list`|
|4|`-n -I test -c '$env.NU_LIB_DIRS'`|`[test]`|`[test]`|
|5|`-n         -c '$env.NU_LIB_DIRS'`|error `column_not_found`|error `column_not_found`|
|6|`-n -I test test.nu`|`[~/.config/nushell/scripts]`|`empty list`|
|7|`-n         test.nu`|`[~/.config/nushell/scripts]`|`empty list`|

Commands and output have been shortened, a default `env.nu` is assumed.

### Expected behavior

The list has at least the values given by the `env.nu` file or the list provided by `-I`.

### Configuration

| key                | value                                                            |
| ------------------ | ---------------------------------------------------------------- |
| version            | 0.83.0                                                           |
| branch             | makepkg                                                          |
| commit_hash        | a33b5fe6ce97b5e9fe8a774c13e783ed65c1b591                         |
| build_os           | linux-x86_64                                                     |
| build_target       | x86_64-unknown-linux-gnu                                         |
| rust_version       | rustc 1.71.0 (8ede3aae2 2023-07-12) (Arch Linux rust 1:1.71.0-1) |
| cargo_version      | cargo 1.71.0                                                     |
| build_time         | 2023-07-26 01:10:19 +00:00                                       |
| build_rust_channel | release                                                          |
| allocator          | standard                                                         |
| features           | dataframe, default, extra, sqlite, trash, which, zip             |
| installed_plugins  |                                                                  |

### Additional context

The issue does not exist in `0.82.0`, but I have not bisected against individual commits.

**Comment by fdncred** [comment]
I think this is because the default_env.nu was changed to not include any NU_LIB_DIRS directories.
https://github.com/nushell/nushell/blob/6ac3351fd13e317aacebaa8203ed01fc94cc4025/crates/nu-utils/src/sample_config/default_env.nu#L73-L74

**Comment by tingerrr** [comment]
Ah OK, well, the config I use is basically default, but it does define the given `NU_LIB_DIRS`.

**Comment by fdncred** [comment]
correct. I think now, to have the same experience, you'd have to do `nu --env-config /path/to/env.nu -c "command | pipeline"`, assuming you have something defined in your `NU_LIB_DIRS` in the env.nu file.

**Comment by tingerrr** [comment]
I figured that's what we have `-I` for, since I really only need the single module for oneoff command execution here. For reference, I stumbled upon this when my keybinds didn't work anymore. I [`sxhkd`](https://github.com/baskerville/sxhkd), where I specify `use mymod.nu; mymod subcommand` as keybinds, `sxhkd` then executes these using `$SHELL -c '...'`. Which no doesn't find the module, so I tried using the `-I` option and came across this discrepancy of behavior.

I'll probably update it to use the `--env-config` option for now.

**Comment by fdncred** [comment]
in order for `nu -c` to work the way it used to the default_env.nu will have to be updated or you'll have to pass `nu --env-config /path/to/your/env.nu -c`

**Comment by tingerrr** [comment]
Ok, maybe I'm misunderstanding something, but I'm trying to understand why row `6` and `7` don't respect the `-I` option the way `4` and `5` do. Despite its description saying `set the NU_LIB_DIRS for the given script`, it doesn't work on the script option.

As well as why `-I` is completely ignored on rows `0` to `3`. If it is intended to be ignored when `-n` is not passed, then it should warn the user that the `-I` option is ignored, or fail. Currently, it seems it simply ignores the option without any indication that it would do so in the help message of either option.

**Comment by fdncred** [comment]
You might try passing the full path in the `-I`. The `-I` was added to support our vscode extension integrations so I've never tried it apart from that. If you have multiple paths with `-I` they're also not semicolon delimited.

This is how they're delimited.
```typescript
"\x1e" + // \x1e is the record separator character (a character that is unlikely to appear in a path)
```

You could also get a full path with `test | path expand` I think.

**Comment by tingerrr** [comment]
I found the mention of `'x1e'` in the source code too, the help message still talks about semicolons for separators which is not quite correct then.

Passing an absolute path that exists does not fix the issue `nu -I /home -c '$env.NU_LIB_DIRS'` still produces an empty list on `0.83.0`, but that was visible in the table I provided. From following the code this should work though.

The issue, from what I can tell, is that this option is evaluated before config files are checked in `run_command`, `run_file` and `run_repl`, only command when using `-n`, completely skips reading any defaults. (Why does this differ between `run_command` and `run_file`, I would expect those two to behave the same here as they are both non-interactive. `run_file` will always load the `default_env.nu` if no file is provided, whereas `run_command` does not. See `src/run.rs`)

As it's used for the IDE functionality, I think the best course of action would be to clear up its documentation and perhaps make the non-interactive versions more consistent. To me, it wasn't clear at all that it's not intended for regular use.

**Comment by amtoine** [comment]
@fdncred 
as long as we keep the two in sync, i'm all for adding this back :yum:

**Comment by spion** [comment]
It seems like `-I` no longer works at all in v0.103

**Comment by spion** [comment]
I see. That does work - `$env.NU_LIB_DIRS` doesn't get populated so I assumed that its was completely ignored.
